subsys/net/lib/download_client/src/download_client.c

```diff
@@ -340,22 +300,74 @@ int socket_send(const struct download_client *client, size_t len)
 	return 0;
 }
 
-static int request_send(struct download_client *dl)
+int poll_timeout(struct download_client *dl)
 {
 	switch (dl->proto) {
 	case IPPROTO_TCP:
 	case IPPROTO_TLS_1_2:
-		return http_get_request_send(dl);
+		return CONFIG_DOWNLOAD_CLIENT_TCP_SOCK_TIMEO_MS;
 	case IPPROTO_UDP:
 	case IPPROTO_DTLS_1_2:
 		if (IS_ENABLED(CONFIG_COAP)) {
-			return coap_request_send(dl);
+			return coap_poll_timeout(dl);
 		}
 	}
 
 	return 0;
 }
 
+static int request_send(struct download_client *dl)
+{
+	/* Request next fragment, if necessary (HTTPS/CoAP) */
+	if (dl->progress == 0 || dl->proto != IPPROTO_TCP ||
+	    IS_ENABLED(CONFIG_DOWNLOAD_CLIENT_RANGE_REQUESTS)) {
+		dl->http.has_header = false;
+
+		switch (dl->proto) {
+		case IPPROTO_TCP:
+		case IPPROTO_TLS_1_2:
+			return http_get_request_send(dl);
+		case IPPROTO_UDP:
+		case IPPROTO_DTLS_1_2:
+			if (IS_ENABLED(CONFIG_COAP)) {
+				return coap_request_send(dl);
+			}
+		}
+	}
+
+	return 0;
+}
+
+static int send_request_again(struct download_client *dl)
+{
+	int err;
+
+	LOG_DBG("Resending request again");
+
+	if (dl->progress == 0 || dl->proto != IPPROTO_TCP ||
+	    IS_ENABLED(CONFIG_DOWNLOAD_CLIENT_RANGE_REQUESTS)) {
+		dl->http.has_header = false;
+
+		switch (dl->proto) {
+		case IPPROTO_TCP:
+		case IPPROTO_TLS_1_2:
+			break;
```

Comment:
thanks!

Reply:
Sry, I had a invalid comment within `send_request_again()`. The code is fine as it is. I will document the code a bit better to avoid misunderstanding :)

Reply:
According to your comment. If a reconnect occurs, poll is waiting for POLLOUT. Once POLLOUT is available, a request is sent. So, everything is fine here.